### ISSUE TITLE
Group item  send only initial data

### ISSUE THREAD

**Issue by agsochi** [jeroenhendricksen/node-red-contrib-openhab3#2]
Hi, if use group Item like this:
```
Group:Number:AVG            Temperatures  "All Room Temperatures [%.1f °C]" // e.g. "21.3 °C"
```
Node receives only initial data, i try all checkboxes in settings.

**Comment by jeroenhendricksen** [comment]
> Hi, if use group Item like this:
> 
> ```
> Group:Number:AVG            Temperatures  "All Room Temperatures [%.1f °C]" // e.g. "21.3 °C"
> ```
> 
> Node receives only initial data, i try all checkboxes in settings.

Good find!
I think I might have removed (or simply not added) support for the group events. I'll see if I can add it.

**Comment by jeroenhendricksen** [comment]
I've fixed support for the `GroupItemStateChangedEvent` event and tested it with additions to the test flow:

<img width="1263" alt="Screenshot 2021-03-10 at 19 47 39" src="https://user-images.githubusercontent.com/32140294/110681042-a4574080-81d9-11eb-9dfc-67e32ef0b4b3.png">

I've release the fix as part of version `1.3.3`. I had to update it manually in my Node-RED because the _Manage palette_ manager didn't see the latest version (yet):
```
$ cd data
$ npm install node-red-contrib-openhab3@latest
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
+ node-red-contrib-openhab3@1.3.3
updated 1 package, moved 1 package and audited 908 packages in 9.659s
```

Please let me know if it solves your issue.

**Comment by agsochi** [comment]
All ok now. Thank you

**Comment by MDAR** [comment]
BUT...

All that said

With 1.3.25, I can see group values 
![image](https://user-images.githubusercontent.com/10905891/141105173-272721d3-7fe2-45e6-8335-2bffaf69cc3e.png)


![image](https://user-images.githubusercontent.com/10905891/141105218-2659011c-8f16-46f3-8d88-2df1fa99fc65.png)


So I will leave it along now

**Comment by jeroenhendricksen** [comment]
> > I've fixed support for the `GroupItemStateChangedEvent` event and tested it with additions to the test flow:
> > <img alt="Screenshot 2021-03-10 at 19 47 39" width="1263" src="https://user-images.githubusercontent.com/32140294/110681042-a4574080-81d9-11eb-9dfc-67e32ef0b4b3.png">
> > I've release the [fix](https://github.com/jeroenhendricksen/node-red-contrib-openhab3/commit/6c3b353482b4ffc75de32c35dd9c9b1d3e85a435) as part of version `1.3.3`. I had to update it manually in my Node-RED because the _Manage palette_ manager didn't see the latest version (yet):
> > ```
> > $ cd data
> > $ npm install node-red-contrib-openhab3@latest
> > npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
> > + node-red-contrib-openhab3@1.3.3
> > updated 1 package, moved 1 package and audited 908 packages in 9.659s
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please let me know if it solves your issue.
> 
> Hi
> 
> I didn't realise I would be using this feature, so I'm delighted that you're already looking into it.
> 
> My Palette Manager isn't offering anything more than 1.3.25 ![image](https://user-images.githubusercontent.com/10905891/141102175-9482fc8b-1486-4d5f-b15e-e231b9f0a019.png) So I tried your command line
> 
> ```
> npm install node-red-contrib-openhab3@latest
> ```
> 
> This spits out a lot of warnings and finally ends with a failure, complaining that PhantomJS isn't available for Linux/Arm64
> 
> I installed phantomJS with Apt and everything seemed to go well, except that `phantomjs --version` returns saying it can't connect to a diplay. so I tried your npm command again.
> 
> ```
> # phantomjs --version
> qt.qpa.screen: QXcbConnection: Could not connect to display
> Could not connect to any X display.
> ```
> 
> This time it still failed, but with with a different error.
> 
> ```
> npm ERR! path /mnt/dietpi_userdata/node-red/node_modules/psl/node_modules/phantomjs-prebuilt
> npm ERR! command failed
> npm ERR! command sh -c node install.js
> npm ERR! Considering PhantomJS found at /usr/bin/phantomjs
> npm ERR! Found PhantomJS at /usr/bin/phantomjs ...verifying
> npm ERR! Error verifying phantomjs, continuing Error: Command failed: /usr/bin/phantomjs --version
> npm ERR! qt.qpa.screen: QXcbConnection: Could not connect to display
> npm ERR! Could not connect to any X display.
> ```
> 
> The Odroid C4 machine doesn't have a monitor attached to it, as it is in a cabinet out of the way. However. it does have a dummy HDMI plug in, so should think it has a display. (Currently QLCplus is running in an xinit session on that display, but no desktop manager)
> 
> This is the highlights from the SSH session I was running to install your update
> 
> ```
> root@openHAB-C4:~# cd /mnt/dietpi_userdata/node-red/node_modules/psl/data
> root@openHAB-C4:/mnt/dietpi_userdata/node-red/node_modules/psl/data# npm install node-red-contrib-openhab3@latest
> npm WARN deprecated har-validator@5.1.5: this library is no longer supported
> npm WARN deprecated debug@3.2.6: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 hav                                                                                                                                                             e a low-severity ReDos regression when used in a Node.js environment. It is reco                                                                                                                                                             mmended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/iss                                                                                                                                                             ues/797)
> npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#dep                                                                                                                                                             recated
> npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older v                                                                                                                                                             ersions may use Math.random() in certain circumstances, which is known to be pro                                                                                                                                                             blematic.  See https://v8.dev/blog/math-random for details.
> npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprec                                                                                                                                                             ated
> npm WARN deprecated request@2.88.2: request has been deprecated, see https://git                                                                                                                                                             hub.com/request/request/issues/3142
> npm WARN deprecated hapi-no-var@1.0.1: This module has moved and is now availabl                                                                                                                                                             e at @hapi/rule-no-var. Please update your dependencies as this version is no lo                                                                                                                                                             nger maintained an may contain bugs and security issues.
> npm WARN deprecated hapi-capitalize-modules@1.1.6: This module has moved and is                                                                                                                                                              now available at @hapi/rule-capitalize-modules. Please update your dependencies                                                                                                                                                              as this version is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated no-arrowception@1.0.0: This module has moved and is now avai                                                                                                                                                             lable at @hapi/rule-no-arrowception. Please update your dependencies as this ver                                                                                                                                                             sion is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated hapi-for-you@1.0.0: This module has moved and is now availab                                                                                                                                                             le at @hapi/rule-for-loop. Please update your dependencies as this version is no                                                                                                                                                              longer maintained an may contain bugs and security issues.
> npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy.                                                                                                                                                              new code should use the URLSearchParams API instead.
> npm WARN deprecated eslint-config-hapi@12.0.0: This module has moved and is now                                                                                                                                                              available at @hapi/eslint-config-hapi. Please update your dependencies as this v                                                                                                                                                             ersion is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated eslint-plugin-hapi@4.1.0: This module has moved and is now a                                                                                                                                                             vailable at @hapi/eslint-plugin-hapi. Please update your dependencies as this ve                                                                                                                                                             rsion is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated hapi-scope-start@2.1.1: This module has moved and is now ava                                                                                                                                                             ilable at @hapi/rule-scope-start. Please update your dependencies as this versio                                                                                                                                                             n is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade                                                                                                                                                              to chokidar 3 with 15x less dependencies.
> npm WARN deprecated phantomjs-prebuilt@2.1.16: this package is now deprecated
> npm notice
> npm notice New major version of npm available! 7.20.6 -> 8.1.3
> npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.1.3
> npm notice Run npm install -g npm@8.1.3 to update!
> npm notice
> npm ERR! code 1
> npm ERR! path /mnt/dietpi_userdata/node-red/node_modules/psl/node_modules/phanto                                                                                                                                                             mjs-prebuilt
> npm ERR! command failed
> npm ERR! command sh -c node install.js
> npm ERR! PhantomJS not found on PATH
> npm ERR! Unexpected platform or architecture: linux/arm64
> npm ERR! It seems there is no binary available for your platform/architecture
> npm ERR! Try to install PhantomJS globally
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2021-11-10T10_56_13_390Z-debug.log
> root@openHAB-C4:/mnt/dietpi_userdata/node-red/node_modules/psl/data# apt update
> Hit:1 https://deb.debian.org/debian buster InRelease
> Get:2 https://deb.debian.org/debian buster-updates InRelease [51.9 kB]
> Get:3 https://deb.debian.org/debian-security buster/updates InRelease [65.4 kB]
> Get:4 https://deb.debian.org/debian buster-backports InRelease [46.7 kB]
> Get:5 https://dietpi.com/meveric all InRelease [19.4 kB]
> Hit:6 https://dietpi.com/meveric buster InRelease
> Hit:7 https://openhab.jfrog.io/artifactory/openhab-linuxpkg stable InRelease
> Get:8 https://deb.debian.org/debian buster-updates/main arm64 Packages.diff/Index [9,100 B]
> Get:9 https://deb.debian.org/debian buster-updates/main arm64 Packages 2021-10-26-2004.40.pdiff [283 B]
> Get:9 https://deb.debian.org/debian buster-updates/main arm64 Packages 2021-10-26-2004.40.pdiff [283 B]
> Get:10 https://deb.debian.org/debian-security buster/updates/main arm64 Packages [303 kB]
> Get:11 https://deb.debian.org/debian buster-backports/contrib arm64 Packages.diff/Index [27.8 kB]
> Get:12 https://deb.debian.org/debian buster-backports/main arm64 Packages.diff/Index [27.8 kB]
> Get:13 https://deb.debian.org/debian buster-backports/contrib arm64 Packages 2021-10-20-1403.22.pdiff [182 B]
> Get:13 https://deb.debian.org/debian buster-backports/contrib arm64 Packages 2021-10-20-1403.22.pdiff [182 B]
> Get:14 https://deb.debian.org/debian buster-backports/main arm64 Packages 2021-10-20-1403.22.pdiff [6,039 B]
> Get:15 https://deb.debian.org/debian buster-backports/main arm64 Packages 2021-10-20-2007.27.pdiff [17.6 kB]
> Get:16 https://deb.debian.org/debian buster-backports/main arm64 Packages 2021-10-27-1402.26.pdiff [76 B]
> Get:17 https://deb.debian.org/debian buster-backports/main arm64 Packages 2021-10-31-0201.34.pdiff [1,389 B]
> Get:17 https://deb.debian.org/debian buster-backports/main arm64 Packages 2021-10-31-0201.34.pdiff [1,389 B]
> Get:18 https://dietpi.com/meveric all/main arm64 Packages [17.5 kB]
> Fetched 594 kB in 6s (106 kB/s)
> Reading package lists... Done
> Building dependency tree
> Reading state information... Done
> 9 packages can be upgraded. Run 'apt list --upgradable' to see them.
> root@openHAB-C4:/mnt/dietpi_userdata/node-red/node_modules/psl/data# apt install phantomjs
> Reading package lists... Done
> Building dependency tree
> Reading state information... Done
> The following NEW packages will be installed:
>   phantomjs
> 0 upgraded, 1 newly installed, 0 to remove and 9 not upgraded.
> Need to get 287 kB of archives.
> After this operation, 959 kB of additional disk space will be used.
> Get:1 https://deb.debian.org/debian buster/main arm64 phantomjs arm64 2.1.1+dfsg-2 [287 kB]
> Fetched 287 kB in 0s (2,085 kB/s)
> Selecting previously unselected package phantomjs.
> (Reading database ... 50541 files and directories currently installed.)
> Preparing to unpack .../phantomjs_2.1.1+dfsg-2_arm64.deb ...
> Unpacking phantomjs (2.1.1+dfsg-2) ...
> Setting up phantomjs (2.1.1+dfsg-2) ...
> Processing triggers for man-db (2.8.5-2) ...
> root@openHAB-C4:/mnt/dietpi_userdata/node-red/node_modules/psl/data# npm install node-red-contrib-openhab3@latest
> npm WARN deprecated eslint-config-hapi@12.0.0: This module has moved and is now available at @hapi/eslint-config-hapi. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
> npm WARN deprecated har-validator@5.1.5: this library is no longer supported
> npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
> npm WARN deprecated hapi-no-var@1.0.1: This module has moved and is now available at @hapi/rule-no-var. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated no-arrowception@1.0.0: This module has moved and is now available at @hapi/rule-no-arrowception. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated hapi-capitalize-modules@1.1.6: This module has moved and is now available at @hapi/rule-capitalize-modules. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated hapi-scope-start@2.1.1: This module has moved and is now available at @hapi/rule-scope-start. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated eslint-plugin-hapi@4.1.0: This module has moved and is now available at @hapi/eslint-plugin-hapi. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated hapi-for-you@1.0.0: This module has moved and is now available at @hapi/rule-for-loop. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
> npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
> npm WARN deprecated debug@3.2.6: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
> npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
> npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
> npm WARN deprecated phantomjs-prebuilt@2.1.16: this package is now deprecated
> npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
> npm ERR! code 1
> npm ERR! path /mnt/dietpi_userdata/node-red/node_modules/psl/node_modules/phantomjs-prebuilt
> npm ERR! command failed
> npm ERR! command sh -c node install.js
> npm ERR! Considering PhantomJS found at /usr/bin/phantomjs
> npm ERR! Found PhantomJS at /usr/bin/phantomjs ...verifying
> npm ERR! Error verifying phantomjs, continuing Error: Command failed: /usr/bin/phantomjs --version
> npm ERR! qt.qpa.screen: QXcbConnection: Could not connect to display
> npm ERR! Could not connect to any X display.
> npm ERR!
> npm ERR!     at ChildProcess.exithandler (node:child_process:397:12)
> npm ERR!     at ChildProcess.emit (node:events:394:28)
> npm ERR!     at maybeClose (node:internal/child_process:1067:16)
> npm ERR!     at Socket.<anonymous> (node:internal/child_process:453:11)
> npm ERR!     at Socket.emit (node:events:394:28)
> npm ERR!     at Pipe.<anonymous> (node:net:672:12) {
> npm ERR!   killed: false,
> npm ERR!   code: 1,
> npm ERR!   signal: null,
> npm ERR!   cmd: '/usr/bin/phantomjs --version'
> npm ERR! }
> npm ERR! Unexpected platform or architecture: linux/arm64
> npm ERR! It seems there is no binary available for your platform/architecture
> npm ERR! Try to install PhantomJS globally
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2021-11-10T11_02_11_868Z-debug.log
> ```
> 
> If you need me to try something, please let me know.

I should have mentioned that when you perform this "manual" package update, and it results in (unrelated to the node-red-contrib-openhab3, AFAI can tell) messages, it's probably best to wait a little while until the palette shows the new version and use the "Update" button in Node-RED to update the package...

**Comment by jeroenhendricksen** [comment]
> BUT...
> 
> All that said
> 
> With 1.3.25, I can see group values ![image](https://user-images.githubusercontent.com/10905891/141105173-272721d3-7fe2-45e6-8335-2bffaf69cc3e.png)
> 
> ![image](https://user-images.githubusercontent.com/10905891/141105218-2659011c-8f16-46f3-8d88-2df1fa99fc65.png)
> 
> So I will leave it along now

Great to hear that it works! Good to know that there's more people using this feature.
